src/goto-programs/remove_returns.cpp

```diff
@@ -46,74 +52,90 @@ class remove_returnst
   void undo_function_calls(
     goto_functionst &goto_functions,
     goto_programt &goto_program);
+
+  symbol_exprt get_or_create_return_value_symbol(
+    const irep_idt &function_id);
 };
 
+symbol_exprt remove_returnst::get_or_create_return_value_symbol(
+  const irep_idt &function_id)
+{
+  const irep_idt symbol_name =
+    id2string(function_id) + RETURN_VALUE_SUFFIX;
+  const symbolt *existing_symbol = symbol_table.lookup(symbol_name);
+  if(existing_symbol != nullptr)
+    return existing_symbol->symbol_expr();
+
+  const symbolt &function_symbol = symbol_table.lookup_ref(function_id);
+  const typet &return_type = to_code_type(function_symbol.type).return_type();
+
+  if(return_type == empty_typet())
+    return symbol_exprt();
+
+  auxiliary_symbolt new_symbol;
+  new_symbol.is_static_lifetime = true;
+  new_symbol.module = function_symbol.module;
+  new_symbol.base_name =
+    id2string(function_symbol.base_name) + RETURN_VALUE_SUFFIX;
+  new_symbol.name = symbol_name;
+  new_symbol.mode = function_symbol.mode;
+  // If we're creating this for the first time, the target function cannot have
+  // been remove_return'd yet, so this will still be the "true" return type:
+  new_symbol.type = return_type;
```

Comment:
Ideally, we should have a consistency check function that could be used to add a debug-mode invariant whether a function has been remove_return'd or not. (Not TBD in  this PR, but a note for future hardening of goto passes.)